tron/utils/__init__.py

```diff
@@ -19,3 +27,47 @@ def next_or_none(iterable):
         return next(iterable)
     except StopIteration:
         pass
+
+
+@contextlib.contextmanager
+def flock(fd):
+    close = False
+    if isinstance(fd, str):
+        fd = open(fd, 'a')
+        close = True
+
+    try:
+        fcntl.lockf(fd, fcntl.LOCK_EX)
```

Comment:
I'm pretty sure you want fcntl.LOCK_EX | fcntl.LOCK_NB' here. With just fcntl.LOCK_EX`, if someone else has locked the file, the process will block until that "someone else" releases the file.

Reply:
fixed